### PR TITLE
Replace clirr with Revapi and enable backwards compatibility checks

### LINKED ISSUE #6499: Enable clirr and backwards compatibility checks after 1.0 release
**Description**

After the 1.0 release we should re-enable clirr and any other backwards compatibility check

### PR DESCRIPTION
**Pull request by npepinpe**
## Description

This PR re-enables backwards compatibility checks post 1.0 in preparation for future releases. To do so, it replaces clirr with [revapi](https://revapi.org).

The main motivations to replace clirr with revapi are the following:

- it's actively maintained (see https://github.com/revapi/revapi)
- it's extensively documented (though it's sometimes hard to navigate, it's all there, and much better than clirr)
- the output it provides is great - it tells you why it failed, where the failure comes from, and how to fix it and/or ignore it
- it has much more powerful matching/filtering features than clirr, which are much simpler to understand/parse than a code list
- it can warn you when you inadvertently expose non-public types via your API (thereby widening the surface of how much you support)

Additionally to the PR, I will also update/rewrite the relevant wiki page explaining how to deal with revapi/compatibility changes.

BREAKING CHANGE: the last commit introduces a breaking change, which are:

1. Make some protected fields private - these fields were exposing "internal" types, e.g. Broker, BrokerCfg, SpringBrokerBridge, types for which we don't want to maintain backwards compatibility guarantees.
2. Remove the `EmbeddedBrokerRule#getBroker()` method - again, exposing the internal `Broker` type.

The reason for this is outlined in the commit, but TL;DR, I want to avoid having to provide compatibility guarantees on types that users aren't meant to use/consume, such as the `Broker`, and by exposing them here, they may start relying on them in their test and expect the same guarantees that the test module offers.

As such, the breaking change is minimal, and serves to prevent future constraints and/or necessary breaking changes.

## Related issues

closes #6499

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

**Comment by menski** [comment]
I will check it out tomorrow, today I'm :exploding_head:

**Comment by npepinpe** [comment]
Yeah no rush :+1:

**Comment by npepinpe** [comment]
After discussion with menski, we decided these breaking changes are small enough that we can go ahead with them. We should communicate them during the release, of course, but they are only breaking if people were using what are really internal/private types.

**Comment by npepinpe** [comment]
One thing to note: Revapi can use both JSON and XML for configuration. I went with JSON which is their default, but I'm happy to switch to XML or try it out if anyone would prefer.

File: test/ignored-changes.json [hunk 1]
```diff
@@ -0,0 +1,57 @@
+[
```
Comment (korthout) [inline test/ignored-changes.json:1]: Nit: all other revapi configuration files are called `revapi.json`.

**Review by korthout — APPROVED** [review]
LGTM 👏  But perhaps others still want to review as well.

It's great that the configuration is moved out of the pom files. Seems to clean up things and there is a stronger focus on justification.

Nothing stood out as unwanted to me.

File: test/ignored-changes.json (hunk 1 above)
Comment (npepinpe) [reply test/ignored-changes.json:1]: I kept this as we previously had `ignored-changes.xml` where we would put the temporarily ignored changes. This file is cleaned out on every release. You can read more about it here: https://github.com/camunda-cloud/zeebe/wiki/Breaking-Changes#java (I will of course update this when we switch to Revapi :sweat_smile:)

**Comment by npepinpe** [comment]
bors merge

**Pull request merged**